src/OneExplorer.ts

```diff
@@ -152,29 +177,131 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     }
   }
 
+  // TODO(dayo) extract file-relative functions as another module
+  private parseInputPath = (configPath: string, modelPath: string): string|undefined => {
+    const config = readIni(configPath);
+    const ext = path.extname(modelPath).slice(1);
+
+    if (ext === undefined || config === null) {
+      return undefined;
+    }
+
+    return config[`one-import-${ext}` as keyof typeof config] ?
+        config[`one-import-${ext}` as keyof typeof config]['input_path'] :
+        undefined;
+  };
+
+  // TODO(dayo) extract file-relative functions as another module
+  private grepTargetInCommand = (str: string): string[] => {
+    let targets: string[] = [];
+    for (let entry of str.split(' ')) {
+      if (path.extname(entry) === '.tvn' || path.extname(entry) === '.circle') {
+        targets.push(entry);
+      }
+    }
+    return targets;
+  };
+
+  // TODO(dayo) extract file-relative functions as another module
+  private grepTarget = (str: string): string[] => {
+    let targets: string[] = [];
+    // TODO(dayo) add checks
+    targets.push(str);
+    return targets;
+  };
+
+  // TODO(dayo) extract file-relative functions as another module
+  private parseIntermediates = (configPath: string): string[] => {
+    const config = readIni(configPath);
+
+    if (config === null) {
+      return [];
+    }
+
+    const targetLocator = [
+      {section: 'one-optimize', key: 'input_path', grepper: this.grepTarget},
+      {section: 'one-optimize', key: 'input_path', grepper: this.grepTarget},
+      {section: 'one-quantize', key: 'input_path', grepper: this.grepTarget},
+      {section: 'one-quantize', key: 'input_path', grepper: this.grepTarget},
+      {section: 'one-codegen', key: 'command', grepper: this.grepTargetInCommand},
+    ];
+
+    let intermediates: string[] = [];
+    for (let loc of targetLocator) {
+      let confSection = config[loc.section as keyof typeof config];
+      let confKey = confSection ? confSection[loc.key as keyof typeof config] : undefined;
```

Comment:
(Optional) Javascript/Typescript has "optional chaining" :D

```suggestion
      let confKey = confSection?.[loc.key as keyof typeof config];
```